index.js

```diff
@@ -1,125 +1,201 @@
+      [t.identifier(localName)]
+    )
+  )
+}
+
+function makeNamespaceObject (t, localName, members) {
+  const properties = members.map(m => t.objectProperty(
+    t.identifier(m.name), t.identifier(`_${localName}_${m.name}`)
+  ))
+  return t.variableDeclaration(
+    'const', [
+      t.variableDeclarator(
+        t.identifier(localName),
+        t.objectExpression(properties)
+      )
+    ]
+  )
 }
 
 module.exports = babelCore => {
   const t = babelCore.types
-
   return {
     visitor: {
-      ImportDeclaration (path, file) {
-        const node = path.node
-        if (!node.source.value.startsWith('glob:') && !glob.hasMagic(node.source.value)) {
+      ImportDeclaration (path, state) {
+        const specifiers = path.node.specifiers
+        const source = path.node.source
+        const error = message => path.buildCodeFrameError(message)
+
+        let pattern = source.value
+        if (pattern.startsWith('glob:')) {
+          pattern = pattern.replace(/^glob:/, '')
+        } else if (!hasMagic(pattern)) {
           return
         }
 
-        rejectImportDefaultSpecifier(path, node.specifiers)
-
-        const fromDir = nodePath.dirname(file.file.opts.filename)
-        const matches = glob.sync(getPattern(path, node.source), {
-          // Search relative to the source file, assuming that location is
-          // derived correctly.
-          cwd: fromDir,
-          strict: true
-        })
-
-        const prefix = commonPathPrefix(matches)
-        const suffix = commonExtname(matches)
-
-        const lookup = Object.create(null)
-        const members = []
-        for (const filepath of matches) {
-          const src = './' + nodePath.relative(fromDir, nodePath.resolve(fromDir, filepath))
-          const subpath = filepath.slice(prefix.length, filepath.length - suffix.length)
-          const pieces = subpath.split(nodePath.sep)
-          const member = pieces.map((name, index) => {
-            const id = identifierfy(name, {
-              prefixReservedWords: pieces.length === 1,
-              prefixInvalidIdentifiers: index === 0
-            })
-            if (id === null) {
-              throw path.buildCodeFrameError(
-                `Could not generate a valid identifier for '${src}'. The '${name}' component could not be converted.`)
-            }
-            return id
-          }).join('$')
-
-          if (lookup[member]) {
-            // hyphen conversion means foo-bar and fooBar will collide.
-            throw path.buildCodeFrameError(`Found colliding members '${member}'`)
-          }
-
-          lookup[member] = src
-          members.push(member)
+        if (hasImportDefaultSpecifier(specifiers)) {
+          throw error('Cannot import the default member')
         }
 
-        if (node.specifiers.length === 0) {
-          path.replaceWithMultiple(members.map(member => t.importDeclaration([], t.stringLiteral(lookup[member]))))
-          return
+        if (!pattern) {
```

Comment:
This would be reached if they just had `"glob:"`. If I add a `hasMagic` when `pattern.startsWith('glob:')` is true, then I can throw an error if pattern is empty or isn't a glob pattern.